composer/utils/object_store/mlflow_object_store.py

```diff
@@ -0,0 +1,392 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""MLFlow Artifacts object store."""
+
+from __future__ import annotations
+
+import logging
+import os
+import pathlib
+import uuid
+from typing import Callable, List, Optional, Tuple, Union
+
+from composer.utils import dist
+from composer.utils.import_helpers import MissingConditionalImportError
+from composer.utils.object_store.object_store import ObjectStore, ObjectStoreTransientError
+
+__all__ = ['MLFlowObjectStore']
+
+MLFLOW_DATABRICKS_TRACKING_URI = 'databricks'
+MLFLOW_DBFS_PATH_PREFIX = 'databricks/mlflow-tracking/'
+
+DEFAULT_MLFLOW_EXPERIMENT_NAME = 'mlflow-object-store'
+
+PLACEHOLDER_EXPERIMENT_ID = 'MLFLOW_EXPERIMENT_ID'
+PLACEHOLDER_RUN_ID = 'MLFLOW_RUN_ID'
```

Comment:
@dakinggg I'll see if I can get the format strings (i.e., `{mlflow_experiment_id}` and `{mlflow_run_id}`) working in the follow-up PR, but I'm finding it pretty tricky to do without needing to rewrite  a lot of the formatting code in `Trainer` and `CheckpointSaver`